src/hooks/useFilterState/useFilterState.stories.mdx

```diff
@@ -0,0 +1,38 @@
+import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';
+import { useFilterState } from './useFilterState';
+import { FilterableTable } from './useFilterState.stories.tsx';
+import GithubLink from '../../../.storybook/helpers/GithubLink';
+
+<Meta title="Hooks/useFilterState" component={useFilterState} />
+
+# useFilterState
+
+A custom, composeable hook for handling filter state.
+
+```ts
+// Syntax:
+const filterState = useFilterState<T>(args: ISelectionStateArgs<T>);
+
+// Arguments:
+interface IFilterStateArgs<T> {
+  items: T[];
+  filterCategories: T[]; // Defaults to []
```

Comment:
filterCategories is required and doesn't have a default

Reply:
Added the default = []  inside the prop destructing. I think that addresses the issue.  Didn't know if i should add a "| undefined"  here or when that would be necessary. I guess we are pretty much always in control of filterCategories so it should never be undefined.